code/modules/computer4/data/c4_file/_c4_file.dm

```diff
@@ -106,8 +106,13 @@
 	if(!origin)
 		origin = containing_folder
 
-	if(!text)
+	// Base cases so we don't need to do any text processing
+	if(!text || text == "." || text == "./") // Current
 		return origin
+	else if(text == "/") // Root
+		return origin.drive.root
+	else if(text == ".." || text == "../") // Parent
+		return origin.containing_folder
```

Comment:
Performance isnt a concern in this path, im more concerned about the directory parser breaking and I don't want shortcuts to hide potential flaws in the parser.